ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx

```diff
@@ -1,58 +1,98 @@
-    deployments: (ListDeployment & { listeningEndpoints: ProcessListeningOnPort[] })[];
-};
-
-function ListeningEndpointsTable({ deployments }: ListeningEndpointsTableProps) {
+function EmbeddedTable({
+    deploymentId,
+    listeningEndpoints,
+}: {
+    deploymentId: string;
+    listeningEndpoints: ProcessListeningOnPort[];
+}) {
     return (
-        <TableComposable borders={false} variant="compact">
+        <TableComposable isNested>
             <Thead noWrap>
                 <Tr>
                     <Th>Program name</Th>
                     <Th>PID</Th>
                     <Th>Port</Th>
                     <Th>Protocol</Th>
-                    <Th>Deployment</Th>
-                    <Th>Namespace</Th>
-                    <Th>Cluster</Th>
                     <Th>Pod ID</Th>
                     <Th>Container name</Th>
                 </Tr>
             </Thead>
             <Tbody>
-                {deployments.length === 0 && (
-                    <Tr>
-                        <Td colSpan={8}>
-                            <Bullseye>
-                                <EmptyStateTemplate title="No results found" headingLevel="h2" />
-                            </Bullseye>
-                        </Td>
+                {listeningEndpoints.map(({ podId, endpoint, signal, containerName }) => (
+                    <Tr key={`${deploymentId}/${podId}/${endpoint.port}`}>
+                        <Td dataLabel="Program name">{signal.name}</Td>
+                        <Td dataLabel="PID">{signal.pid}</Td>
+                        <Td dataLabel="Port">{endpoint.port}</Td>
+                        <Td dataLabel="Protocol">{l4ProtocolLabels[endpoint.protocol]}</Td>
+                        <Td dataLabel="Pod ID">{podId}</Td>
+                        <Td dataLabel="Container name">{containerName}</Td>
                     </Tr>
-                )}
-                {deployments.flatMap(({ id, name, namespace, cluster, listeningEndpoints }) =>
-                    listeningEndpoints.map(({ podId, endpoint, signal, containerName }) => (
-                        <Tr key={`${id}/${podId}/${endpoint.port}`}>
-                            <Td dataLabel="Program name">{signal.name}</Td>
-                            <Td dataLabel="PID">{signal.pid}</Td>
-                            <Td dataLabel="Port">{endpoint.port}</Td>
-                            <Td dataLabel="Protocol">{l4ProtocolLabels[endpoint.protocol]}</Td>
+                ))}
+            </Tbody>
+        </TableComposable>
+    );
+}
+
+export type ListeningEndpointsTableProps = {
+    deployments: (ListDeployment & { listeningEndpoints: ProcessListeningOnPort[] })[];
+};
+
+function ListeningEndpointsTable({ deployments }: ListeningEndpointsTableProps) {
+    const expandedRowSet = useSet<string>();
+    return (
+        <TableComposable variant="compact">
+            <Thead noWrap>
+                <Tr>
+                    <Th>{/* Header for expanded column */}</Th>
```

Comment:
Let’s ask at Developer Office Hours tomorrow what it recommended idiom to prevent accessibility issue from axe DevTools for empty table head cell.

Reply:
Ah good call, we're following the same pattern in other places in the app as well so if we find a more accessible solution we should update each place if possible.

Reply:
@dvail Consensus on the call was replace `<Th />` with `<Td />` which I did somewhere else.

Although Nicole mentioned `aria-label` attribute, we have 3 `<Th aria-label="Row actions" />` for which axe DevTools reports the minor issue for empty heading cell text.
https://github.com/patternfly/patternfly-design/issues/1176

Reply:
Hmm interesting. I wonder if that is an error in the axe tools, or if aria-label really doesn't function sufficiently for screen readers.

Either way though, the Th->Td fix is quick and I can include it in the next PR.

Reply:
I think the Axe tool takes a pretty strict stance on table headers, and intentional requires a visible header (not just a screen reader version), because it makes the purpose of things like a "kebab/actions" column clear to the most number of users.